site/functions/upload.ts

```diff
@@ -1,14 +1,33 @@
+import { count } from 'drizzle-orm';
+import { drizzle } from 'drizzle-orm/d1';
+import { images } from '../schema';
+
 interface Env {
   BUCKET: R2Bucket;
   BOARD_WEBHOOKS: string;
+  DB: D1Database;
 }
 
 export const onRequest: PagesFunction<Env> = async (context) => {
+  const db = drizzle(context.env.DB);
   const [stream1, stream2] = context.request.body.tee();
   const filename = new Date().toISOString() + '.png';
+  const numImages = (await db.select({ value: count() }).from(images))?.[0]
+    ?.value;
+  if (!numImages) {
+    const imgs = (await context.env.BUCKET.list()).objects;
+    const values = imgs
+      .filter((img) => img.key.startsWith('202'))
+      .map((img) => ({
+        path: img.key,
+        created_at: img.uploaded,
+      }));
+    await db.insert(images).values(values);
+  }
   await Promise.all([
     await context.env.BUCKET.put('latest.png', stream1),
     await context.env.BUCKET.put(filename, stream2),
+    await db.insert(images).values({ path: filename, created_at: new Date() }),
   ]);
```

Comment:
We need to add new images to the D1 database, so we now insert on the upload.

We also need a way to backfill all the images already added so we:
1. check if the db is empty
2. query everything in the bucket and upload it to D1

This should only run when the first image uploads. We can remove these lines in a following PR.